chore: add changelog from 13.6.3 (#5607)
cypress-io/cypress-documentation: docs/guides/references/changelog.mdx

title: Changelog
---

## 13.6.3

_Released 1/16/2024_

**Bugfixes:**

> - Force `moduleResolution` to `node` when `typescript` projects are detected to correctly run Cypress. This change should not have a large impact as `commonjs` is already forced when `ts-node` is registered. This fix does not impact the ESM Typescript configuration loader. Fixes [#27731](https://github.com/cypress-io/cypress/issues/27731).

Reviewer: ```suggestion
- Force [`moduleResolution`](https://www.typescriptlang.org/tsconfig#moduleResolution) to `node` when `typescript` projects are detected. This change should have minimal impact as `commonjs` is already forced for [`module`](https://www.typescriptlang.org/tsconfig#module) when `ts-node` is registered. This fix does not impact the ESM Typescript configuration loader. Fixes [#27731](https://github.com/cypress-io/cypress/issues/27731).
```

- Fixed an issue where some cross-origin logs, like assertions or cy.clock(), were getting too many dom snapshots. Fixes [#28609](https://github.com/cypress-io/cypress/issues/28609).
- Fixed asset capture for Test Replay for requests that are routed through service workers. This addresses an issue where styles were not being applied properly in Test Replay and [`cy.intercept()`](/api/commands/intercept) was not working properly for requests in this scenario. Fixes [#28516](https://github.com/cypress-io/cypress/issues/28516).
- Fixed an issue where visiting an `http://` site would result in an infinite reload/redirect loop in Chrome 114+. Fixes [#25891](https://github.com/cypress-io/cypress/issues/25891).
- Fixed an issue where requests made from extra tabs do not include their original headers. Fixes [#28641](https://github.com/cypress-io/cypress/issues/28641).
- Fixed an issue where [`cy.wait()`](/api/commands/wait) would sometimes throw an error reading a property of undefined when returning responses. Fixes [#28233](https://github.com/cypress-io/cypress/issues/28233).

> **Performance:**

Reviewer 2: we want to list performance before bug fixes typically

Author: really? I always thought it was always the other way around.

Reviewer 2: Idk if it matters much but: https://github.com/cypress-io/cypress/blob/develop/guides/writing-the-cypress-changelog.md#writing-guidelines
